ffi/src/lib.rs

```diff
@@ -1755,3 +1776,275 @@ pub(crate) unsafe fn get_pipeline_options(
+                    .unwrap_or_else(|_| {
+                        CString::new("Couldn't convert error message to C string").unwrap()
+                    })
+                    .into_raw();
+            }
+        }
+    }
+
+    // Initialize open telemetry metrics exporter
+    if open_telemetry_config.has_metrics {
+        let metrics = open_telemetry_config.metrics;
+        let endpoint = unsafe { CStr::from_ptr(metrics.endpoint).to_string_lossy() };
+        match GlideOpenTelemetrySignalsExporter::from_str(&endpoint) {
+            Ok(exporter) => {
+                config = config.with_metrics_exporter(exporter);
+            }
+            Err(e) => {
+                let error_msg = format!("Invalid metrics exporter configuration: {}", e);
+                return CString::new(error_msg)
+                    .unwrap_or_else(|_| {
+                        CString::new("Couldn't convert error message to C string").unwrap()
+                    })
+                    .into_raw();
+            }
+        }
+    }
+
+    let flush_interval_ms = if open_telemetry_config.has_flush_interval_ms {
+        open_telemetry_config.flush_interval_ms
+    } else {
+        DEFAULT_FLUSH_SIGNAL_INTERVAL_MS as i64
+    };
+
+    if flush_interval_ms <= 0 {
+        let error_msg = format!(
+            "InvalidInput: flushIntervalMs must be a positive integer (got: {})",
+            flush_interval_ms
+        );
+        return CString::new(error_msg)
+            .unwrap_or_else(|_| CString::new("Couldn't convert error message to C string").unwrap())
+            .into_raw();
+    }
+
+    config = config.with_flush_interval(std::time::Duration::from_millis(flush_interval_ms as u64));
+
+    // Initialize OpenTelemetry synchronously
+    match ensure_tokio_runtime() {
+        Ok(runtime) => {
+            match runtime.block_on(async { GlideOpenTelemetry::initialise(config.build()) }) {
+                Ok(_) => std::ptr::null(), // Success
+                Err(e) => {
+                    let error_msg = format!("Failed to initialize OpenTelemetry: {}", e);
+                    CString::new(error_msg)
+                        .unwrap_or_else(|_| {
+                            CString::new("Couldn't convert error message to C string").unwrap()
+                        })
+                        .into_raw()
+                }
+            }
+        }
+        Err(e) => CString::new(e)
+            .unwrap_or_else(|_| CString::new("Couldn't convert error message to C string").unwrap())
+            .into_raw(),
+    }
+}
+
+/// Frees a C string.
+///
+/// # Safety
+/// * `s` must be a valid pointer to a C string.
+#[unsafe(no_mangle)]
+pub unsafe extern "C" fn free_c_string(s: *mut c_char) {
+    unsafe {
+        if s.is_null() {
+            return;
+        }
+        drop(CString::from_raw(s));
+    };
+}
+
+fn ensure_tokio_runtime() -> Result<&'static Runtime, String> {
+    static TOKIO: OnceLock<Runtime> = OnceLock::new();
```

Comment:
We have a way to get the runtime using `glide_core::client::get_or_init_runtime` already, which is already static. You should use that instead.